### PR TITLE
Fix iOS Camera Texture memory leak on dispose

### PR DESCRIPTION
**Pull request by Alex-MSFT**
The current implementation of the cameraTextureDelegate was taking a strong reference to the implData object, which was preventing the destructor from being run on camera close.  This PR adds a small cleanup protocol to cleanly handle deallocation, and removes the circular reference.

Before:
Screen Shot 2022-09-13 at 10 39 43 AM

After:
Screen Shot 2022-09-13 at 11 50 13 AM

### CODE REVIEW

**Review by CoPrez — APPROVED** [review]

File: Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm
Comment (docEdub) [inline Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:606]: Extra whitespace here

File: Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm
Comment (docEdub) [inline Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:473]: We should probably wait to make this until it's needed

Comment (docEdub) [reply Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:473]: I'll leave it up to you. It's fine by me either way.

Comment (Alex-MSFT) [reply Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:473]: I think I'll just leave it since the Open call assumes we have a valid pointer

Comment (docEdub) [inline Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:473]: We should probably change `m_implData` back to `std::unique_ptr` in the header now that it's not shared anymore.

Comment (Alex-MSFT) [reply Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:473]: Good call!

File: Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm
Comment (ryantrem) [inline Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:353]: Does `dispatch_get_global_queue` have a specific thread affinity? I took a quick look at the docs and couldn't tell. Overall what is this `dispatch_async` call trying to do?

Comment (Alex-MSFT) [reply Plugins/NativeCamera/Source/Apple/NativeCameraImpl.mm:353]: The goal is to run this on any background thread, as running it on the main thread will hold up app responsiveness.  In general the important one to offload is stopRunning as that will lock up the app for ~3 seconds if run on the main thread, but a warning is generated when running this on the main thread as well.

**Pull request merged**
